### PR TITLE
Completeness and other tweaks

### PR DESCRIPTION
**Pull request by alice-hannah**
* Completeness table added to appendix and completeness text automated in report.
* Use custom headers with page breaks
* Add date stamp to open data files
* Formatting tweaks to figures

### CODE REVIEW

**Comment by alice-hannah** [comment]
I'll deal with the merge conflicts once you have reviewed and approved ☝️👍

File: functions/completeness.R [hunk 1]
```diff
@@ -20,64 +18,54 @@ completeness <- function(end_date, level = c("board", "scotland")) {
...
-                  readr::parse_number(financial_year) == 
+                  readr::parse_number(quarter) == 
                     lubridate::year(end_date) - 1) %>%
     dplyr::left_join(lookup, by = "hb2014") %>%
     dplyr::rename(board = hb2014name) %>%
     dplyr::mutate(board = replace(board,
                                   hb2014 == "S92000003",
                                   "Scotland")) %>%
-    tidyr::drop_na(board)
+    tidyr::drop_na(board) %>%
```
Comment (Tina815) [inline functions/completeness.R:59]: I noticed you excluded SB0801 (The Golden Jubilee National Hospital). The special board lookup can be found at https://www.opendata.nhs.scot/dataset/non-standard-geography-codes-and-labels/resource/0450a5a2-f600-4569-a9ae-5d6317141899 

You have this hospital in the publication report. So we should have "The Golden Jubilee National Hospital" in "board" before drop_na(board). Need to do the same thing for fy.

File: markdown/summary.Rmd [hunk 2]
```diff
@@ -57,7 +56,7 @@ Note that hospital data and National Records of Scotland deaths data are provisi
...
+* In `r latest$fy`, there were `r latest$deaths` deaths in Scotland excluding those where an external cause of death was recorded (including falls).  For these individuals, `r latest$qom`% of their last six months of life was spent either at home or in a community setting, with the remaining `r latest$qom_hosp`% spent in hospital. This is equivalent to an average of `r if(latest$hosp < 10){as.character(as.english(latest$hosp))}else{latest$hosp}` days in hospital in the six months prior to death.
 
 * Over the past `r as.character(as.english(year(end_date) - year(start_date)))` years, the percentage of time spent at home or in a community setting has gradually increased from `r summarise_data(basefile, include_years = "first") %>% pull(qom)`% in `r min(basefile$fy)` to `r latest$qom`% in `r latest$fy`.
```
Comment (Tina815) [inline markdown/summary.Rmd:62]: I can't comment for the background as it was approved. We've discussed it before. For this statement:

"The percentage of time spent at home or in a community setting towards the end of life provides a high level indication of progress in implementation of the national action plan."

Do you agree to change "national action plan" to "strategic framework objectives"? As strategic framework superseded the national action plan. I see the text is changed in \\nssstats01\irf\18-End-of-Life\Publication\EOL publication May 2018\2018-05-29-End-of-Life-Summary.docx

File: markdown/report.Rmd [hunk 3]
```diff
@@ -521,27 +554,42 @@ More recently, the December 2016 the ["Health and Social Care Delivery Plan"](ht
...
 
 It is envisaged that an increase in this measure will reflect both quality and value through more effective, person centred and efficient end of life care with people being better able to be cared for at home or closer to home with a planned approach to end of life care resulting in less time in a hospital setting. 
 
 Over time, an increase in the proportion of time spent at home / in a community setting at the end of life may indicate an improvement in the quality and efficiency of palliative care. This increase would be expected in the medium to long term as a result of an increase in the use of anticipatory care plans and electronic palliative care summaries. 
 
 This measure is based on Acute Inpatient and Day Case and Geriatric Long Stay (SMR01/SMR01 GLS) and Mental Health Inpatient and Day Case (SMR04), returns covering non-obstetric hospitals in Scotland. 
 
+<div custom-style = "Heading 3_pgbrk">
 **Data Quality and Completeness**
```
Comment (Tina815) [inline markdown/report.Rmd]: Don't need to have ** as Heading 3_pgbrk will have the bold text anyway?

File: markdown/report.Rmd [hunk 4]
```diff
@@ -204,7 +237,7 @@ paste("All data are revised to ensure the most complete data are presented. Data
...
+paste("SMR01 hospital completeness for the financial year", latest$fy, "is estimated to be", comp_scot, "complete and this could have an impact on the time spent in hospital in the last six months of life.", comp_metadata, "See [Appendix 2](#appendix-2---background-information) for more information and completeness figures for each quarter. Further updates  will be made available at the following link:  [http://www.isdscotland.org/Products-and-Services/Data-Support-and-Monitoring/SMR-Completeness/](http://www.isdscotland.org/Products-and-Services/Data-Support-and-Monitoring/SMR-Completeness/)"),
 paste("Due to a change in methodology, results presented in this publication will not be comparable to figures published prior to August 2016, Trend information from 2010/11 to", latest$fy, "is available in this publication."),
 "It is the policy of ISD Scotland to make its web sites and products accessible according to published guidelines.",
 "Publication is made available via the ISD website.",
```
Comment (Tina815) [inline markdown/report.Rmd:243]: Can't leave a comment for next published date as it has been approved. I noticed you use "next_pub_date" for Next published. I think it's correct, although by having a look at the May and October 2019 publication they are both the date of current publication (e.g. for May 2019 publication, the Next published is 28/05/2019) - just want to double check which date you want to use?

File: code/02_create-figures.R [hunk 5]
```diff
@@ -109,16 +111,16 @@ fig2 <-
...
            fontface = 2) +
   annotate("text", 
            x = 4e+05, y = 1100000, 
            label = paste0("NHS ", unique(fig2$HBName[which.max(fig2$qom)]),
                           ": ", max(round_half_up(fig2$qom, 1)), "%"), 
-           size = 3.5,
+           size = 2,
            fontface = 2)
 
 ggsave(here("markdown", "figures", "figure-2.png"),
```
Comment (Tina815) [inline code/02_create-figures.R]: I think the current width and height make the annotation too small after embedding into the report. After testing can we set the width = 13 and height = 14? I think that looks better.

**Comment by Tina815** [comment]
In report.Rmd "Age and Gender" section, it should be "Table 4 in the [data tables]" instead of Table "3" as I've added Council Area to the excel tables, so that makes Age Gender as Table 4, Deprivation as Table 5, Urban Rural as Table 6. 

So for "Deprivation" section, we need to modify the text to "Table 5 in the [data tables]". For "Urban Rural Classification" section, need to modify the text to "Table 6 in [data tables]".

Also I noticed the hyperlink in the round bracket should be (\`r data_tables\`) from Table 2 onwards (so Table 1 is correct). Now they are (data_tables) which will make the hyperlinks not work.

File: markdown/report.Rmd [hunk 6]
```diff
@@ -90,44 +143,24 @@ contact <- data.frame(contact1, contact2, contact3)
...
-  ))
+
+file_name <- dir_ls(here("data", "excel-output"), regexp = pub_date)
+file_size <- paste0("Excel ", file_size(file_name), "b")
+file_desc <- c(paste0("Percentage of last six months of life spent at home or ",
+                      "in the community setting by financial year for each ",
+                      "Health Board, HSCP, Council Area, Age, Sex, ",
+                      "Deprivation and Rurality"),
+               "Figures and Background Data")
+file_link <- c(data_tables, figures)
```
Comment (Tina815) [inline markdown/report.Rmd]: We need to reverse the order of the elements for file_desc and file_link, as from file_name 2020-05-28_figures.xlsm is the first item and 2020-05-28_qom.xlsm is the second item.

File: markdown/summary.Rmd (hunk 2 above)
Comment (alice-hannah) [reply markdown/summary.Rmd:62]: Sorry Tina - I remember we discussed this and I never make the change - I agree and have changed in summary and report background sections.

File: markdown/report.Rmd (hunk 4 above)
Comment (alice-hannah) [reply markdown/report.Rmd:243]: How weird! - I've not noticed this before. I'll double check with Ag, but I definitely think this is correct as is and should be date of next publication.

File: code/02_create-figures.R (hunk 5 above)
Comment (alice-hannah) [reply code/02_create-figures.R]: Thank Tina - I've tweaked size and increased font size to 2.5 and looks better now. I've also added the following bit of code to the word macro to centre all images. 

For Each oILShp In ActiveDocument.InlineShapes
oILShp.Select
Selection.ParagraphFormat.Alignment = wdAlignParagraphCenter
Next

File: functions/completeness.R (hunk 1 above)
Comment (alice-hannah) [reply functions/completeness.R:59]: Thanks for spotting Tina - I've added this now. I did have a problem reading the special health board lookup using `ckanr::ckan_fetch()` so I've switched to using `read_csv` for all.

**Comment by alice-hannah** [comment]
Hi Tina,
Thanks for your review - I've made the changes required and also updated the excel template with your new version with appendix trend charts. 

I've also made changes suggested by Stefan to open data script. 

As I said, once you're happy with the changes and have approved I'll do something about the merge conflicts 🤞

**Comment by Tina815** [comment]
Great, thanks Alice! I will start having a look tomorrow!

File: code/04_create-open-data.R [hunk 7]
```diff
@@ -133,6 +133,13 @@ agesex <-
                                 fy == max(.$fy),
                               "p", "")) %>%
   
+  mutate(age = case_when(
+    str_detect(age, "-") ~ paste(age, "years"),
+    str_detect(age, "\\+") ~ paste(age, "years and over")
```
Comment (Tina815) [inline code/04_create-open-data.R]: Shall we delete the "+" sign as it's already "85 years and over"?

File: functions/completeness.R [hunk 8]
```diff
@@ -10,74 +8,82 @@ completeness <- function(end_date, level = c("board", "scotland")) {
...
-      
-    }
-    
-  }
+  fy <- readr::read_csv(paste0("https://www.opendata.nhs.scot/dataset/",
+                               "110c4981-bbcc-4dcb-b558-5230ffd92e81/",
+                               "resource/daf55fd2-457f-4845-9af1-5d154cc0b19c",
+                               "/download/financialyr.csv")) %>%
+    janitor::clean_names() %>%
+    dplyr::filter(smr_type == "SMR01",
```
Comment (Tina815) [inline functions/completeness.R:69]: Shall we also filter is.na(completeness_qf) as qtr does? I know it doesn't matter as eventually you left joined fy to qtr, but just want to keep things tidy?

**Review by Tina815 — APPROVED** [review]

**Pull request merged**